CPS-0STL/README.md

```diff
@@ -0,0 +1,158 @@
+2. A smart-contract transaction escrows the funds/tokens on the partner chain.
+3. Simultaneously, a mirror of that smart-contract transaction is submitted on Cardano.
+4. After a short amount of time, the Cardano transaction has been incorporated into a newly-formed block.
+5. Wait for settlement or non-settlement, either by waiting for a specific number of blocks to cover the transaction or by observing some other aspect of the chain.
+	1. If the transaction settled, complete the escrow contract on the partner chain.
+	2. If the transaction was rolled back, resubmit it.
+
+> [!NOTE]
+> Write out detailed use cases, preferably contributed by stakeholders.
+
+
+## Goals
+
+The overall goals are to precisely define settlement metrics and to propose protocol or parameter changes that significantly speed settlement without negatively impacting performance or security.
+
+1. Develop precise settlement/finality metrics relevant for stakeholders.
+	1. Some metrics should be defined in terms of the whole life cycle of a transaction, from its submission to the memory pool to its becoming settled in a block.
+	2. Block- versus slot-based metrics have importance for different use cases.
+	3. Embody these metrics in an open-source library or toolkit for estimating settlement times or measuring finality of Cardano transactions.
+	4. Create metrics that are practical for wallets to implement and to present clearly to users.
+2. Shorten settlement time, as defined by stakeholder-relevant metrics, in the face of even moderate adversarial activity.
+	1. Stake-based adversaries.
+	2. Attacks utilizing adversarial resources (CPU or network).
+	3. Natural disruption of infrastructure or networks (e.g., data-center or internet outages).
+4. Fall back to Praos-like security in the face of strongly adversarial conditions.
+
+It addition to the goals above, it is advisable to avoid the following potentially costly changes.
+
+1. Avoid making major changes to the existing Ouroboros protocol parameters or logic.
+2. Do not weaken Ouroboros security or substantially enlarge its attack surface.
+3. Minimize changes that increase the resource usages of Cardano nodes or the cost of operating them.
+
+Three approaches are under active consideration or development to address the settlement and finality problem.
+
+- *Voting approaches* strengthen the weight of the preferred chain, making it extremely difficult to roll back blocks on the prefix of the chain that was made weightier by a quorum (consensus) of stake-based votes. [Ouroboros Peras](https://github.com/cardano-foundation/CIPs/pull/872) is an example of this. This approach does not mitigate CPU-resource attacks.
+- *Anti-grinding approaches* make prohibitively expensive any attacks that rely on CPU resources to weaken cryptographic guarantees of the pseudo-randomness of the slot-leadership schedule. This approach does not mitigate stake-based attacks.
```

Comment:
Maybe that is a matter of definition of "stake-based attack", but nonce grinding can also be seen as a at least partially stake-based attack as an adversary needs stake in order to decide how to best use its elected slots to influence the nonce. E.g. https://eprint.iacr.org/2021/1698.pdf mentions that in the synchronous setting, adversaries with less than 9.5% of the stake have inherently limited grinding power without even having to resort to CPU resource assumptions.

Reply:
I guess for the sake of simplicity and clarity we want to distinguish two broad classes of solutions mitigating 2 broad classes of attacks?

Reply:
Added clarification that grinding requires stake, too, in a4960f0328f5ecc7e1c2ab75211a5e83f272ba98.